addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -530,12 +530,20 @@ function factory(dependencies) {
             // 1. move messages from inbox to history
             // AKU TODO: flag other caches to invalidate
             // task-2171873
-            inboxMailbox.messages.map(message => {
-                message.update({
-                    isNeedaction: false,
-                    isHistory: true,
-                });
-            });
+            for (const message_id of message_ids) {
+                // We need to ignore all not yet known messages because we don't want them
+                // to be shown partially as they would be linked directly to mainCache
+                // Furthermore, server should not send back all message_ids marked as read
+                // but something like last read message_id or something like that.
+                // (just imagine you mark 1000 messages as read ... )
```

Comment:
Don't forget to create task-id. (don't need to cancel merge to add it, but make sure task is created)

Reply:
Thing is "mark all as read" makes server responsible in which messages are marked as read. If we omit reception of message_ids in the longpolling notification, we may mistakenly move a message from inbox to history and introduce an inconsistency.

So it's annoying that we have to provide whole list of message_ids, but I think it's a necessary evil.

Reply:
We don't need to omit it, we need just need to return the last id.

Btw this method has the same issue as channel_seen, in that any new message between the click and the response would be incorrectly(?) marked as read too. The method could take a param to know which is the last one at the time of the request.